pytorch_lightning/strategies/deepspeed.py

```diff
@@ -68,24 +68,21 @@ def remove_module_hooks(model: torch.nn.Module) -> None:
 
 class LightningDeepSpeedModule(_LightningModuleWrapperBase):
     def __init__(
-        self, pl_module: Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase], precision: int
+        self, pl_module: Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase], precision: Union[str, int]
     ) -> None:
         super().__init__(pl_module)
         self.precision = precision
 
     def forward(self, *inputs, **kwargs):
         if self.precision == 16:
-            inputs = self._move_float_tensors_to_half(inputs)
-
+            inputs = self._move_float_tensors_to(inputs, torch.Tensor.half)
+        elif self.precision == PrecisionType.BFLOAT:
+            inputs = self._move_float_tensors_to(inputs, torch.Tensor.bfloat16)
         return super().forward(*inputs, **kwargs)
 
     @staticmethod
-    def batch_to(data):
-        return data.half()
-
-    def _move_float_tensors_to_half(self, batch: Any):
-        batch = apply_to_collection(batch, (torch.FloatTensor, torch.cuda.FloatTensor), function=self.batch_to)
-        return batch
+    def _move_float_tensors_to(batch: Any, function: Callable[[torch.Tensor], torch.Tensor]) -> Any:
+        return apply_to_collection(batch, (torch.FloatTensor, torch.cuda.FloatTensor), function=function)
```

Comment:
Wouldn't it be safer to use:

https://pytorch.org/docs/stable/generated/torch.is_floating_point.html

instead of the explicit classes?